Mage.Client/src/main/java/mage/client/cards/DragCardGrid.java

```diff
@@ -2020,7 +2020,7 @@ private void notifyCardsSelected() {
     private void selectCard(CardView targetCard) {
         // Set the selected card to the target card
         for (CardView card : allCards) {
-            if (card == targetCard) {
+            if (Objects.equals(card, targetCard)) {
```

Comment:
You are replace with `equals`, but `CardImpl` and maybe other objects (need research) hasn't overrided equals method. You must implement it too (see #4407 or `PlayerImpl` as example).

![shot_220704_214444](https://user-images.githubusercontent.com/8344157/177200118-9ce8d8fb-2f5b-4d1c-8770-fd07129bf192.png)

Reply:
It's about game engine, not GUI elements (GUI uses refs only in most use cases).

Reply:
> It's about game engine objects, not GUI elements (GUI uses refs only in most use cases).

I thought it was good practice to compare all objects using their `.equals()` method. Prevents bugs by not requiring the devs to think if this object can be compared with == or if it needs .equals.

Reply:
Yes, that's right. I'm talking about the idea from original issue: search code for `==` to find potential objects with miss equals implementation and usage.

Reply:
I will go through and implement .equals() for whichever classes need it.